### ISSUE TITLE
Dashboard exception details not visible in app host console

### ISSUE THREAD

**Issue by JamesNK** [dotnet/aspire#4697]
Dashboard error logs are in the app host console, but exception details aren't visible.

![image](https://github.com/dotnet/aspire/assets/303201/2fc22872-55e7-4f8e-9f14-6928265701a1)

What is in dashboard:

```
Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost: Error: Unhandled exception in circuit 'NjBWFS143AYAUetpcO5oF0ZFkNRrD5zhBFQpfb9tB4c'.

System.InvalidOperationException: Can't get ApplicationKey from resource type details 'Type = Singleton, InstanceId = apigateway-jqp2h8hqfo, ReplicaSetName = ' because ReplicaSetName is null.
   at Aspire.Dashboard.Model.ResourceTypeDetails.GetApplicationKey() in C:\Development\Source\aspire\src\Aspire.Dashboard\Model\ResourceTypeDetails.cs:line 28
   at Aspire.Dashboard.Components.Pages.StructuredLogs.UpdateViewModelFromQuery(StructuredLogsPageViewModel viewModel) in C:\Development\Source\aspire\src\Aspire.Dashboard\Components\Pages\StructuredLogs.razor.cs:line 339
   at Aspire.Dashboard.Components.Pages.PageExtensions.InitializeViewModelAsync[TViewModel,TSerializableViewModel](IPageWithSessionAndUrlState`2 page) in C:\Development\Source\aspire\src\Aspire.Dashboard\Components\Pages\IPageWithSessionAndUrlState.cs:line 87
   at Aspire.Dashboard.Components.Pages.StructuredLogs.OnParametersSetAsync() in C:\Development\Source\aspire\src\Aspire.Dashboard\Components\Pages\StructuredLogs.razor.cs:line 136
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
```

cc @davidfowl

**Comment by davidfowl** [comment]
https://github.com/dotnet/aspire/blob/74c12fd1671bb25465ce5479dcc59a9df46cd24d/src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs#L319

😄

**Comment by JamesNK** [comment]
IMO this is important to fix in 8.1 so people understand what is happening when the dashboard doesn't work. The logs currently provided aren't good enough when the exception is missing.
